src/ucp/core/ucp_context.h

```diff
@@ -184,8 +213,16 @@ typedef struct ucp_context {
         /* Configuration supplied by the user */
         ucp_context_config_t      ext;
 
+        /* Thread mode of current context */
+        ucs_thread_mode_t         thread_mode;
+
     } config;
 
+    /* Lock for multithreading support */
+    pthread_mutex_t               thread_mutex;
```

Comment:
pls do union between mutex and spinlock to document the fact only one of them is used